R/trace_generation.R

```diff
@@ -41,13 +41,24 @@ layer2traces <- function(l, d, misc) {
   
   ## For non-numeric data on the axes, we should take the values from
   ## the original data.
-  for (axis.name in c("x", "y")){
-    if (!misc$is.continuous[[axis.name]]){
+  for (axis.name in c("x", "y")) {
+    if (!misc$is.continuous[[axis.name]]) {
       aes.names <- paste0(axis.name, c("", "end", "min", "max"))
       aes.used <- aes.names[aes.names %in% names(g$aes)]
       for(a in aes.used) {
         col.name <- g$aes[aes.used]
-        data.vec <- l$data[[col.name]]
+        dtemp <- l$data[[col.name]]
+        if (is.null(dtemp)) {
+          if (!inherits(g$data[[paste0(a, ".name")]], "NULL")) {
+            # Handle the case where as.Date() is passed in aes argument.
+            if (class(g$data[[a]]) != class(g$data[[paste0(a, ".name")]])) {
+              g$data[[a]] <- g$data[[paste0(a, ".name")]]
+              data.vec <- g$data[[a]]
+            }
+          }
```

Comment:
Yes, well, if `data.vec` is not set, it's `NULL` and it's still okay because I `try()` in subsequent operations (https://github.com/ropensci/plotly/blob/master/R/trace_generation.R#L56, etc.).  So the exceptions are handled.

My initial intention was to leave line 50
```r
data.vec <- l$data[[col.name]]
```
and follow with
```r
if (is.null(data.vec)) {
  if (!inherits(g$data[[paste0(a, ".name")]], "NULL") && (class(g$data[[a]]) != class(g$data[[paste0(a, ".name")]]))) {
    g$data[[a]] <- g$data[[paste0(a, ".name")]]
    data.vec <- g$data[[a]]
  }
}
```
The `else` being, well, leave it alone (it will remain `NULL`).  Do you think this would convey more meaning?

The two `if`s really mean one condition (i.e., the two columns are of different non-null classes) but I split them for readability, also to save one check if the first condition is not even met.